README.adoc

```diff
@@ -0,0 +1,69 @@
+
+Once you've fulfilled the <<Requirements>>, you can use the code located here to provision/manage this infrastructure:
+
+* Configure the backend (storing the shared state for terraform) with a local file named `backend-config`
+** This file has a "properties file format" (e.g. "key=value" per line) to configure the link:https://www.terraform.io/docs/language/settings/backends/s3.html[Terraform S3 Backend storing state]
+** For the production, the content can be retrieved from the outputs of the link:{private_repo_url}[(private) repository {private_repo_name}]
+** This file (`backend-config`) is git-ignored
+
+* Use one the following make commands to execute the associated actions:
+** Run `make help` to print the Makefile manual with all available targets (useful when this `README` is not synced with the `Makefile`)
+** Run `make lint` to statically lint the project resources. It fails if you didn't.
+** Run `make lint-tests` to statically lint the test resources. It fails if you didn't.
+** Run `make prepare` to initialize the local Terraform project (e.g. generates `.terraform` and retrieve dependencies, using the S3 backend)
+** Run `make validate` to run a static analysis of the project (both Terraform and terratest code)
+** Run `make test` to execute the test harness on the project (uses the terratest code in `./tests/`)
+** Run `make plan` to prints the changes planned to be applied on the infrastructure (see this step as a "dry run" of `make deploy`)
+** Run `make deploy` to apply the changes to the infrastructure (implies terraform apply with no approval)
+
+=== Pipeline's Workflows
+
+The `Jenkinsfile_k8s` pipeline describes the workflows for this GitOps system.
+
+There 4 possible workflows:
+
+* When on a Pull Request, then the tests are run on the staging environment AND a terraform plan is run on the production environment (to give the maintainer an idea of what should happen if the PR is merged)
+* When on the principal branch:
+** By default (e.g. a code push, mostly from PR merged), a terraform plan is run on the production AND a deployment is done if the plan succeeds.
+** If the trigger is the daily cron, then all tests are run on the staging AND a terraform plan is run on the production. Please note that link:https://github.com/jenkins-infra/digitalocean/blob/0cba6160b0d8459176cd7c92400171f93efd5ae9/Jenkinsfile_k8s#L92-L94[if the plan detects changes], then it fails because a configuration drift happened.
+** If the trigger is manual, then no tests are run at all, a terraform plan is run on the production and a human must approve the pipeline for the deployment to production to happen
+
+== Troubleshoot
+
+If you've got this error:
+
+> Error: error configuring S3 Backend: error validating provider credentials: error calling sts:GetCallerIdentity: ExpiredToken: The security token included in the request is expired
+
+You'll need to refresh you AWS token.
```

Comment:
Not sure how to explain it here as I'm using a proprietary cli tool to update the token.

Reply:
WDYT about "you need a valid AWS CLI credential" with
- Link to the AWS CLI credential configuration documentation
- a snippet to test the credentials (eg. `aws s3 ls` that should have the bucket defined for backend)